sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/test/TestTableProvider.java

```diff
@@ -157,16 +169,18 @@ public BeamTableStatistics getTableStatistics(PipelineOptions options) {
     public PCollection<Row> buildIOReader(
         PBegin begin, BeamSqlTableFilter filters, List<String> fieldNames) {
       PCollection<Row> withAllFields = buildIOReader(begin);
-      if (fieldNames.isEmpty() && filters instanceof DefaultTableFilter) {
+      if (options == PushDownOptions.NONE) { // needed for testing purposes
         return withAllFields;
       }
 
```

Comment:
In this function you should throw an exception if the filter is set but filter push-down isn't enabled. Same thing with project push down.

Reply:
Added checks that throw a `RuntimeException` when an invalid scenario is encountered.